lib/dialects/sqlite3/query/compiler.js

```diff
@@ -12,25 +12,24 @@ const {
   identity,
 } = require('lodash');
 
-function QueryCompiler_SQLite3(client, builder) {
-  QueryCompiler.call(this, client, builder);
-
-  const { returning } = this.single;
+class QueryCompiler_SQLite3 extends QueryCompiler {
 
-  if (returning) {
-    this.client.logger.warn(
-      '.returning() is not supported by sqlite3 and will not have any effect.'
-    );
-  }
-}
+  constructor(client, builder) {
+    super(client, builder);
 
-inherits(QueryCompiler_SQLite3, QueryCompiler);
+    const { returning } = this.single;
 
-assign(QueryCompiler_SQLite3.prototype, {
-  // The locks are not applicable in SQLite3
-  forShare: emptyStr,
+    if (returning) {
+      this.client.logger.warn(
+        '.returning() is not supported by sqlite3 and will not have any effect.'
+      );
+    }
+    const emptyStr = constant('');
```

Comment:
Do we need to redefine this on every call? Maybe this could be bumped to the top of the file to only be invoked once?

Reply:
Ok, updated.